sdk/src/main/java/com/uid2/sdk/network/DefaultNetworkSession.kt

```diff
@@ -0,0 +1,75 @@
+package com.uid2.sdk.network
+
+import com.uid2.sdk.extensions.decodeJsonToMap
+import java.net.HttpURLConnection
+import java.net.URL
+import java.net.URLEncoder
+
+/**
+ * A default implementation of NetworkSession that leverages HttpUrlConnection to make the necessary
+ * GET and POST requests.
+ */
+open class DefaultNetworkSession : NetworkSession {
+
+    override fun loadData(url: URL, request: NetworkRequest): Map<String, Any> {
+        val connection = openConnection(url).apply {
+            requestMethod = request.type.toRequestMethod()
+
+            setRequestProperty("Accept", "application/json")
+
+            // If we've been given any request headers, add them to the request.
+            request.headers.forEach {
+                addRequestProperty(it.key, it.value)
+            }
+
+            // If data was provided, this will require us to write it to the output stream.
+            request.data?.let { data ->
+                doOutput = true
+
+                outputStream.use { outputStream ->
+                    outputStream.write(data.toPostData().toByteArray(Charsets.UTF_8))
+                }
+            }
+        }
+
+        // A successful response code should be in the [200-299] range.
+        val responseCode = connection.responseCode
+        if (responseCode !in SUCCESS_CODE_RANGE_MIN..SUCCESS_CODE_RANGE_MAX) {
+            return mapOf()
```

Comment:
Are we sure we want to return an empty map for an error condition?  How do we detect the difference between this and the server giving us an empty map in response?  Do we need to?

Reply:
I mentioned in the comment associated with the `NetworkSession` interface, but I will be revisiting this as I integrate a consumer of it. I've already made a start on this with some changes locally, and exploring improved failure cases and what better to return. I suggest this is okay for the initial PR